### ISSUE TITLE
Make FFImageLoading Linker-safe

### ISSUE THREAD

**Issue by akamud** [luberda-molinet/FFImageLoading#1317]
### Description

FFImageLoading uses `Preserve=All` in all classes AFAIK. I understand the reason, but there is no way for a developer to force Linker to remove unused classes that has been marked with `Preserve`, and since FFImageLoading has many features, paying for all of them just for using a single feature, like `CachedImage`, may be too much.

My suggestion: make everything Linker Safe and add the linker XML node on README when someone does not feel like dealing with linker issues. That way both cases can be addressed.

I can help with both changes: making it all LinkerSafe and providing the initial catch-all rule for Linker for people who don't care about its app size for some reason.

What do you think?

**Comment by akamud** [comment]
@daniel-luberda is this something you want fixed? I can send a PR.

**Comment by daniel-luberda** [comment]
@akamud Yes, that would be great, but you'll need to base on `next` branch

**Comment by Tommigun1980** [comment]
Hi @daniel-luberda. Will you accept the pull request any day soon?

Thanks.
